cmd/influx/cli/cli.go

```diff
@@ -322,8 +322,41 @@ func (c *CommandLine) use(cmd string) {
 		return
 	}
 	d := args[1]
-	c.Database = d
-	fmt.Printf("Using database %s\n", d)
+
+	// validate if specified database exists
+	query := "SHOW DATABASES"
+	response, err := c.Client.Query(client.Query{Command: query})
+	if err != nil {
+		fmt.Printf("ERR: %s\n", err)
+		return
+	}
+
+	if err := response.Error(); err != nil {
+		fmt.Printf("ERR: %s\n", err)
+		return
+	}
+
+	// verify the provided database exists
+	databaseExists := false
+	for _, result := range response.Results {
```

Comment:
I suggest a minor improvement:

```
databaseExists := func() bool {
for _, result := range response.Results {
	for _, row := range result.Series {
		if row.Name == "databases" {
			for _, values := range row.Values {
				for _, database := range values {
					if database == d {
						return true
					}
				}
			}
		}
	}
    return false
}()
```
Using a local anonymous function like this allows you to stop looking the moment you find the database.